lib/psci/psci_setup.c

```diff
@@ -96,7 +96,7 @@ static void __init psci_update_pwrlvl_limits(void)
 				psci_non_cpu_pd_nodes[nodes_idx[j]].cpu_start_idx
 					= cpu_idx;
 			}
-			psci_non_cpu_pd_nodes[nodes_idx[j]].ncpus++;
+			psci_non_cpu_pd_nodes[SPECULATION_SAFE_VALUE(nodes_idx[j])].ncpus++;
```

Comment:
Again, I'm struggling to see the threat as this is also in the cold boot path.

Reply:
As discussed offline, we have revised decisions around cold boot path.